array_library.js

```diff
@@ -89,21 +85,40 @@ const createMethod = (method) => {
     }
     return this._isChainable ? this : result;
   };
-};
+}
 
-var ArrayLib = function (currentValue) {
-  this.take = createMethod(take);
-  this.skip = createMethod(skip);
-  this.map = createMethod(map);
-  this.reduce = createMethod(reduce);
-  this.filter = createMethod(filter);
-  this.foreach = createMethod(foreach);
-  this.chain = chain;
-  this.value = value;
-  this._isChainable = false;
-  this._currentValue = currentValue;
-  return this;
-};
+class ArrayLib {
+  constructor(currentValue) {
+    this._isChainable = false;
+    this._currentValue = currentValue;
+  }
+
+  value() {
+    return this._currentValue;
+  }
+
+  take(...args) {
+    return createMethod(take).call(this, ...args);
+  }
+  skip(...args) {
+    return createMethod(skip).call(this, ...args);
+  }
+  map(...args) {
+    return createMethod(map).call(this, ...args);
+  }
+  reduce(...args) {
+    return createMethod(reduce).call(this, ...args);
+  }
+  filter(...args) {
+    return createMethod(filter).call(this, ...args);
+  }
+  foreach(...args) {
+    return createMethod(foreach).call(this, ...args);
+  }
+  chain(obj) {
+    return chain(obj);
```

Comment:
You can write your chain code inside class method and remove the external function :)